From aa7e4a52ecd68efe4ce79ffd1fde5c0c8c7ca851 Mon Sep 17 00:00:00 2001
From: Jeroen Hofstee <jhofstee@victronenergy.com>
Date: Wed, 15 Jan 2025 16:08:19 +0100
Subject: [PATCH 2/2] fix the Sec-WebSocket-Protocol

chat and the typo char are example protocols. The actual protocol is binary.

Upstream-Status: Submitted [https://github.com/any1/neatvnc/pull/141]
---
 src/stream/ws/handshake.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/stream/ws/handshake.c b/src/stream/ws/handshake.c
index 85b1b3e..8088118 100644
--- a/src/stream/ws/handshake.c
+++ b/src/stream/ws/handshake.c
@@ -72,7 +72,7 @@ ssize_t ws_handshake(char* output, size_t output_maxlen, const char* input)
 	bool have_protocols = strlen(protocols) != 1;
 	bool have_versions = strlen(versions) != 1;
 
-	if (have_protocols && !strstr(protocols, ",chat,"))
+	if (have_protocols && !strstr(protocols, ",binary,"))
 		goto failure;
 
 	if (have_versions && !strstr(versions, ",13,"))
@@ -97,7 +97,7 @@ ssize_t ws_handshake(char* output, size_t output_maxlen, const char* input)
 			"%s%s"
 			"\r\n",
 			response,
-			have_protocols ? "Sec-WebSocket-Protocol: char\r\n" : "",
+			have_protocols ? "Sec-WebSocket-Protocol: binary\r\n" : "",
 			have_versions ? "Sec-WebSocket-Version: 13\r\n" : "");
 
 	ssize_t header_len = req.header_length;
-- 
2.43.0

